### PR TITLE
Fix random test failure

### PR DESCRIPTION
**Pull request by bmwiedemann**
While working on reproducible builds for openSUSE, I noticed
that test_popen_nice would randomly fail on 1-core-VMs

```
         remotenice = gw.remote_exec(getnice).receive()
         gw.exit()
         if remotenice is not None:
             gw = makegateway("popen//nice=5")
             remotenice2 = gw.remote_exec(getnice).receive()
 >           assert remotenice2 == remotenice + 5
 E           assert 0 == (0 + 5)
```

### CODE REVIEW

**Comment by bmwiedemann** [comment]
rebased to master and added more sleeps to make tests pass.
Maybe instead there should be some code change so that gw.remote_foo waits a bit for the status to settle, to make the whole library easier to use?
Does not need to be constant waiting times, but could wait for some "update done" message

**Comment by bmwiedemann** [comment]
Often the behaviour of 1-core-VMs can be simulated by prefixing a command with `taskset 1`

**Comment by nicoddemus** [comment]
I don't really know `execnet` internals in order to evaluate the changes, but the tests pass and the change is in the test suite only... @RonnyPfannschmidt @hpk42 should we merge this as is, or does someone wants to take the time to implement the "update done" mechanism proposed by @bmwiedemann?

**Comment by RonnyPfannschmidt** [comment]
i'm no longer working on execnet

File: testing/test_gateway.py [hunk 1]
```diff
@@ -90,6 +91,7 @@ def test_gateway_status_busy(self, gw):
         ch2.waitclose()
         for i in range(10):
             status = gw.remote_status()
```
Comment (hpk42) [inline testing/test_gateway.py]: ``gw.remote_status()`` is a synchronous operation -- no time.sleep() should be neccessary.

Comment (bmwiedemann) [reply testing/test_gateway.py]: but it was already repeated 10 times - so it is waiting on something.

Comment (bmwiedemann) [reply testing/test_gateway.py]: See also the test run from the first commit:
https://travis-ci.org/pytest-dev/execnet/jobs/427581580

Comment (RonnyPfannschmidt) [reply testing/test_gateway.py]: i believe with the introduction of the execmodel backends behaviour changed the expectations and a wait-for with some sleep time is needed to accommodate for the extra sleep time

**Comment by hpk42** [comment]
``gw.remote_status()`` does not change the dict it returns, afterwards. 

Do i see it correctly that all test failures are bout "nice" settings and verifications? 

If so, i'd say there is some flakyness there and the time.sleep() just make it less likely issues occur.

File: testing/test_gateway.py [hunk 2]
```diff
@@ -363,6 +365,7 @@ def test_status_with_threads(self, makegateway):
         # is already closed.
         for i in range(10):
             rstatus = gw.remote_status()
+            time.sleep(0.1)
             if rstatus.numexecuting == 0:
                 return
```
Comment (hpk42) [inline testing/test_gateway.py]: i'd do the time.sleep() as the last command in the for-body, i.e. after the "return". 
i think we could merge this one then.

File: testing/test_xspec.py [hunk 3]
```diff
@@ -137,6 +138,7 @@ def getnice(channel):
         gw.exit()
         if remotenice is not None:
             gw = makegateway("popen//nice=5")
+            time.sleep(0.1)
             remotenice2 = gw.remote_exec(getnice).receive()
             assert remotenice2 == remotenice + 5
```
Comment (hpk42) [inline testing/test_xspec.py]: do you have a supporting traceback for changing this line?

Comment (bmwiedemann) [reply testing/test_xspec.py]: It is part of the first commit's message.

Comment (bmwiedemann) [reply testing/test_xspec.py]: I uploaded a full log at https://www.zq1.de/~bernhard/temp/python-execnet-build.log
To trigger it, I generated some parallel load with `perl -e 'fork;fork;fork;while(1){}'`

**Comment by bmwiedemann** [comment]
Added 4th sleep because of https://travis-ci.org/pytest-dev/execnet/jobs/432459677

**Comment by bmwiedemann** [comment]
ping. Anything left to do?

**Comment by nicoddemus** [comment]
Thanks for the ping @bmwiedemann, I'll let @hpk42 decide on this one.

**Comment by hpk42** [comment]
Can any of these test failures (without your sleep's) be reproduced on python3? 

it's

**Comment by bmwiedemann** [comment]
@hpk42 https://travis-ci.org/pytest-dev/execnet/jobs/432459677 from above said `py35`
and [the one I can reproduce locally](https://www.zq1.de/~bernhard/temp/python-execnet-build.log) talks about `py.test-3.6`

so I think it is a general timing problem.
I'm not sure if this could be problematic outside of tests - in that case, adding `@flakytest` would not be the right change.

**Comment by hpk42** [comment]
thanks for the clarification.  i don't see a difference between @flakytest as compared to time.sleep -- in fact silently passing a timing problem (time.sleep) is worse than marking a test and getting it reported as "x" or "X".

**Comment by bmwiedemann** [comment]
Updated to use `@flakytest` but could only fully test the `test_xspec` part that I can reproducible locally.

**Pull request merged**